### PR TITLE
Moving some code to GameViewModel class

### PR DESCRIPTION
**Pull request by rezneac**
Moving some code to another class so that when changing the configuration of the application (screen rotation) some data is saved and can be used.

### CODE REVIEW

File: app/src/main/java/com/example/android/guesstheword/screens/game/GameViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    var word = ""
+
+    // The current score
+    var score = 0
+
+    // The list of words - the front of the list is the next word to guess
+    private lateinit var wordList: MutableList<String>
+
+    init {
+        Log.i("GameViewModel", "GameViewModel created!")
```
Comment (fdcboris) [inline app/src/main/java/com/example/android/guesstheword/screens/game/GameViewModel.kt:18]: Please delete Logs

File: app/src/main/java/com/example/android/guesstheword/screens/game/GameViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        wordList.shuffle()
+    }
+
+    /**
+     * Moves to the next word in the list
+     */
+    private fun nextWord() {
+        //Select and remove a word from the list
+        if (wordList.isEmpty()) {
+//            gameFinished()
```
Comment (fdcboris) [inline app/src/main/java/com/example/android/guesstheword/screens/game/GameViewModel.kt:64]: Please do not leave commented code

File: app/build.gradle [hunk 3]
```diff
@@ -58,4 +58,6 @@ dependencies {
     // Navigation
     implementation "android.arch.navigation:navigation-fragment-ktx:1.0.0-rc02"
     implementation "android.arch.navigation:navigation-ui-ktx:1.0.0-rc02"
+
+    implementation 'androidx.lifecycle:lifecycle-extensions:2.0.0'
```
Comment (fdcboris) [inline app/build.gradle:62]: Can't see where you're using this dependencies

**Review by fdcboris — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/android/guesstheword/screens/game/GameViewModel.kt (hunk 1 above)
Comment (rezneac) [reply app/src/main/java/com/example/android/guesstheword/screens/game/GameViewModel.kt:18]: In the next pull request I will delete the logs

File: app/build.gradle (hunk 3 above)
Comment (rezneac) [reply app/build.gradle:62]: I used this dependency to see how the life cycle of some classes works.

**Pull request merged**